st2common/st2common/runners/paramiko_ssh.py

```diff
@@ -123,6 +123,7 @@ def __init__(self, hostname, port=DEFAULT_SSH_PORT, username=None, password=None
 
         self.bastion_client = None
         self.bastion_socket = None
+        self.socket = None
```

Comment:
This is the only place I can find where `self.socket` is assigned, which means that almost none of your other changes are being executed. Do you mind explaining a bit more about this fix?

Reply:
@blag This works by update based on your below fix https://github.com/StackStorm/st2/pull/4881#discussion_r401992283